python/packages/nisar/workflows/ionosphere.py

```diff
@@ -551,10 +812,76 @@ def insar_ionosphere_pair(original_cfg, runw_hdf5):
+
+                phase_second = pathlib.Path(orig_scratch_path,
+                                            'ionosphere',
+                                            'main_diff_ms_band',
+                                            'RIFG.h5')
+            else:
+                phase_second = out_paths['RIFG']
+
+            diff_phase_output = pathlib.Path(diff_dir, 'RIFG.h5')
+            iono_insar_cfg['product_path_group'][
+                'scratch_path'] = diff_dir
+            iono_insar_cfg['product_path_group'][
+                'sas_output_file'] = f'{diff_dir}/RUNW.h5'
+            iono_insar_cfg['processing']['dense_offsets'][
+                'coregistered_slc_path'] = new_scratch
+            iono_insar_cfg['processing']['crossmul'][
+                'coregistered_slc_path'] = new_scratch
+            _, out_paths = h5_prep.get_products_and_paths(iono_insar_cfg)
+            os.makedirs(diff_dir, exist_ok=True)
+            shutil.copy(phase_second, diff_phase_output)
+            shutil.copy(additional_runw, out_paths['RUNW'])
+
+            pol_list_a = iono_freq_pols['A']
+            pol_list_b = iono_freq_pols['B']
+            swath_path = RIFGGroupsPaths().SwathsPath
+            first_data_path = []
+            for pol_a in pol_list_a:
+
+                dest_freq_path = f"{swath_path}/frequencyA"
+                dest_pol_path = f"{dest_freq_path}/interferogram/{pol_a}"
+                rifg_path_freq = f"{dest_pol_path}/wrappedInterferogram"
+
+                first_data_path.append(rifg_path_freq)
+            first_slant_path = f"{dest_freq_path}/interferogram/slantRange"
+            second_data_path = []
+            for pol_b in pol_list_b:
+
+                dest_freq_path = f"{swath_path}/frequencyB"
+                dest_pol_path = f"{dest_freq_path}/interferogram/{pol_b}"
+                rifg_path_freq = f"{dest_pol_path}/wrappedInterferogram"
+
+                second_data_path.append(rifg_path_freq)
+            second_slant_path = f"{dest_freq_path}/interferogram/slantRange"
+
+            output_data_path = second_data_path
+            compute_differential_phase(phase_first,
+                                       phase_second,
+                                       diff_phase_output,
+                                       first_data_path,
+                                       second_data_path,
+                                       output_data_path,
+                                       iono_args['lines_per_block'],
+                                       first_slant_path=first_slant_path,
+                                       second_slant_path=second_slant_path)
+
+            # Since main_diff_low_high_subband method does not need to
+            # unwrap low and high subband interferogram, but need to
+            # unwrap the difference between low and high subband interferogram
+            unwrap.run(iono_insar_cfg, out_paths['RIFG'], out_paths['RUNW'])
```

Comment:
Are you creating a new RIFG whose phase is the diff of low and high , only to unwrap? why didn't you use any simpler raster for this intermediate data?